### PR TITLE
Handle source 404 in portfolio item detail.

### PR DESCRIPTION
**Pull request by Hyperkid123**
https://projects.engineering.redhat.com/browse/SSP-1302
### Changes
- when the portfolio item source is no longer available, show a warning alert and disable the order button

screenshot-ci foo redhat com_1337-2020 03 05-16_54_38

### CODE REVIEW

**Comment by gmcculloug** [comment]
@sbuenafe-rh Are you still looking into getting the text reviewed for this case and when the sources is unavailable?  Any ETA on that?

**Comment by sbuenafe-rh** [comment]
@gmcculloug was following up on it with other apps. If this is the Catalog Admin and/or the Catalog User, would they be able to view Sources to rectify the situation?

We can merge the PR and file a bug to get the text updated.

**Comment by Hyperkid123** [comment]
@sbuenafe-rh this message will appear if the source no longer exists. It was deleted from Sources. I don't know if there is hard or soft (archive) delete.

**Comment by gmcculloug** [comment]
@sbuenafe-rh Today, only org_admins on the platform can make changes to sources.  RBAC permissions do not effect this at all.

File: src/smart-components/portfolio/portfolio-item-detail/portfolio-item-detail.js [hunk 1]
```diff
@@ -77,15 +78,24 @@ const PortfolioItemDetail = () => {
...
-              availability={source.availability_status}
+              availability={availability}
             />
+            {source.notFound && (
+              <Alert
+                className="pf-u-ml-lg pf-u-mr-lg"
+                id="source-unavailable-alert-info"
+                variant="warning"
+                isInline
+                title="Platform for this product is no longer available"
```
Comment (lgalis) [inline src/smart-components/portfolio/portfolio-item-detail/portfolio-item-detail.js]: The title here should match the one in line 98

Comment (Hyperkid123) [reply src/smart-components/portfolio/portfolio-item-detail/portfolio-item-detail.js]: These are actually different cases. This platform is just deleted, but the alert bellow has an existing source but its status is `unavailable`. Susan is working someone on the terminology.

**Comment by lgalis** [comment]
@Hyperkid123 - The last commit handling the 404 for non-existent portfolios - that was for PR 448, not this one, right? For this PR - just adding the 'The' in the front of the title was the only change left.

**Comment by Hyperkid123** [comment]
Yup, I've mixed them up. :sleepy:

**Pull request merged**